### PR TITLE
fahclient: 7.6.21 -> 8.1.18

### PR DESCRIPTION
**Pull request by zimbatm**
## Description of changes

Rebuild from source to work around the python2 issue.

Weirdly, https://foldingathome.org/start-folding/?lng=en-US is stuck at v7.6.21.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/misc/foldingathome/client.nix [hunk 1]
```diff
@@ -1,60 +1,61 @@
...
 
   meta = {
     description = "Folding@home client";
     homepage = "https://foldingathome.org/";
-    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
-    license = lib.licenses.unfree;
+    license = lib.licenses.gpl3;
+    mainProgram = "fah-client";
     maintainers = [ lib.maintainers.zimbatm ];
-    platforms = [ "x86_64-linux" ];
```
Comment (drupol) [inline pkgs/applications/science/misc/foldingathome/client.nix:58]: How about leaving this ?

File: pkgs/applications/science/misc/foldingathome/client.nix [hunk 2]
```diff
@@ -1,60 +1,66 @@
...
+
+    cp -r --no-preserve=mode ${cbangSrc} $CBANG_HOME
+  '';
+
+  preBuild = ''
+    scons -C $CBANG_HOME
+  '';
+
+  installPhase = ''
+    mkdir -p $out/{bin,share/applications,share/feh-client}
```
Comment (drupol) [inline pkgs/applications/science/misc/foldingathome/client.nix:50]:
```suggestion
    runHook preInstall

    mkdir -p $out/{bin,share/applications,share/feh-client}
```

File: pkgs/applications/science/misc/foldingathome/client.nix [hunk 3]
```diff
@@ -1,60 +1,66 @@
...
-  ] ++ extraPkgs;
+    cp fah-client $out/bin/fah-client
 
-  runScript = "/bin/FAHClient";
+    cp install/lin/fah-client.desktop $out/share/applications/
+    cp -r images $out/share/feh-client/
 
-  extraInstallCommands = ''
-    mv $out/bin/$name $out/bin/FAHClient
+    sed -e "s|Icon=.*|Icon=$out/share/feh-client/images/fahlogo.png|g" -i $out/share/applications/fah-client.desktop
```
Comment (drupol) [inline pkgs/applications/science/misc/foldingathome/client.nix:57]:
```suggestion
    sed -e "s|Icon=.*|Icon=$out/share/feh-client/images/fahlogo.png|g" -i $out/share/applications/fah-client.desktop
    
    runHook postInstall
```

**Review by drupol — CHANGES_REQUESTED** [review]

**Review by drupol — APPROVED** [review]
LGTM now! Thanks for fixing the remaining things

**Comment by zimbatm** [comment]
Regarding the platforms, I think it should work on aarch64, but I haven't tested it. Anyways, let's get this merged and iterate on it as needed.

**Comment by w4** [comment]
This change allows fah-client to boot and retrieve work, however the application impurely pulls in a worker client at runtime which fails to start, it's not immediately clear as to why:

```
$ pwd
/var/lib/foldingathome/cores/fahcore-a8-lin-64bit-avx2_256-0.0.12

$ ./FahCore_a8
-bash: ./FahCore_a8: cannot execute: required file not found

$ ldd FahCore_a8
	linux-vdso.so.1 (0x00007ffe687bc000)
	libpthread.so.0 => /nix/store/qn3ggz5sf3hkjs2c797xf7nan3amdxmp-glibc-2.38-27/lib/libpthread.so.0 (0x00007f9857196000)
	libdl.so.2 => /nix/store/qn3ggz5sf3hkjs2c797xf7nan3amdxmp-glibc-2.38-27/lib/libdl.so.2 (0x00007f9857191000)
	libm.so.6 => /nix/store/qn3ggz5sf3hkjs2c797xf7nan3amdxmp-glibc-2.38-27/lib/libm.so.6 (0x00007f98570b1000)
	libc.so.6 => /nix/store/qn3ggz5sf3hkjs2c797xf7nan3amdxmp-glibc-2.38-27/lib/libc.so.6 (0x00007f9856ec9000)
	/lib64/ld-linux-x86-64.so.2 => /nix/store/qn3ggz5sf3hkjs2c797xf7nan3amdxmp-glibc-2.38-27/lib64/ld-linux-x86-64.so.2 (0x00007f985719d000)
```

**Comment by zimbatm** [comment]
@w4: can you test this PR^. I'm not running FAH at the moment.

**Pull request merged**